src/qvimagecore.cpp

```diff
@@ -232,10 +234,11 @@ QFileInfoList QVImageCore::getCompatibleFiles()
     const auto &regs = qvApp->getFilterRegExpList();
     const auto &mimeTypes = qvApp->getMimeTypeNameList();
 
-    const QFileInfoList currentFolder = currentFileDetails.fileInfo.dir().entryInfoList();
-    for (const QFileInfo &fileInfo : currentFolder)
+    QDirIterator it(currentFileDetails.fileInfo.absoluteDir().path(), QDir::Filter::Files);
+    while (!it.next().isEmpty())
     {
```

Comment:
I'm not sure why but using `QDirIterator` directly (instead of `entryInfoList()` which relies on `QDirIterator` behind the scenes) resulted in a significant performance increase when testing on Windows. On Mac, the `entryInfoList()` call was already fast to begin with, and this change didn't affect performance. I passed in Filter::Files because the existing code could add folders to `fileInfoList`, for example if a folder was named like "test.jpg".